web/components/contract/contract-details.tsx

```diff
@@ -120,74 +127,157 @@ export function ContractDetails(props: {
+  const { contract, isMobile, disabled } = props
   const groupToDisplay = getGroupLinkToDisplay(contract)
   const groupInfo = groupToDisplay ? (
     <Link prefetch={false} href={groupPath(groupToDisplay.slug)}>
       <a
         className={clsx(
-          linkClass,
-          'flex flex-row items-center truncate pr-0 sm:pr-2',
+          'flex flex-row items-center truncate pr-1',
           isMobile ? 'max-w-[140px]' : 'max-w-[250px]'
         )}
       >
-        <UserGroupIcon className="mx-1 inline h-5 w-5 shrink-0" />
-        <span className="items-center truncate">{groupToDisplay.name}</span>
+        <div className="bg-greyscale-4 hover:bg-greyscale-3 text-2xs items-center truncate rounded-full px-2 text-white sm:text-xs">
+          {groupToDisplay.name}
+        </div>
       </a>
     </Link>
   ) : (
-    <Button
-      size={'xs'}
-      className={'max-w-[200px] pr-2'}
-      color={'gray-white'}
-      onClick={() => !groupToDisplay && setOpen(true)}
+    <Row
+      className={clsx(
+        'cursor-default select-none items-center truncate pr-1',
+        isMobile ? 'max-w-[140px]' : 'max-w-[250px]'
+      )}
     >
-      <Row>
-        <UserGroupIcon className="mx-1 inline h-5 w-5 shrink-0" />
-        <span className="truncate">No Group</span>
-      </Row>
-    </Button>
+      <div
+        className={clsx(
+          'bg-greyscale-4 text-2xs items-center truncate rounded-full px-2 text-white sm:text-xs'
+        )}
+      >
+        No Group
+      </div>
+    </Row>
   )
-
   return (
-    <Row className="flex-1 flex-wrap items-center gap-2 text-sm text-gray-500 md:gap-x-4 md:gap-y-2">
-      <Row className="items-center gap-2">
-        <Avatar
-          username={creatorUsername}
-          avatarUrl={creatorAvatarUrl}
-          noLink={disabled}
-          size={6}
-        />
-        {disabled ? (
-          creatorName
-        ) : (
-          <UserLink
-            className="whitespace-nowrap"
-            name={creatorName}
-            username={creatorUsername}
-            short={isMobile}
-          />
-        )}
-        {!disabled && <UserFollowButton userId={creatorId} small />}
-      </Row>
-      <Row>
+    <>
+      <Row className="align-middle">
         {disabled ? (
-          groupInfo
-        ) : !groupToDisplay && !user ? (
-          <div />
+          { groupInfo }
         ) : (
+          // !user ? (
+          //   <div />
+          // ) :
           <Row>
             {groupInfo}
-            {user && groupToDisplay && (
-              <Button
-                size={'xs'}
-                color={'gray-white'}
+            {user && (
+              <button
+                className="text-greyscale-4 hover:text-greyscale-3"
                 onClick={() => setOpen(!open)}
               >
-                <PencilIcon className="mb-0.5 mr-0.5 inline h-4 w-4 shrink-0" />
-              </Button>
+                <PlusCircleIcon className="mb-0.5 mr-0.5 inline h-4 w-4 shrink-0" />
```

Comment:
Thanks, this icon is wayyy better than the pencil here.

Reply:
Thank you! :)